res/controllers/engine-api.d.ts

```diff
@@ -31,6 +31,24 @@ declare interface ScriptConnection {
     readonly isConnected: boolean;
 }
 
+/** QmlPlayerProxy */
+declare interface PlayerManager {
+    isLoaded(): boolean;
+    getTrack(): string;
+    getTitle(): string;
```

Comment:
I don't think these are accessible via JS because those functions are not `Q_INVOKABLE`. You should however be able to directly access the declared `Q_PROPERTY`s. So just `console.log(player.title)` for example. 

Also slight misnaming, this is not the playermanager:
```suggestion
declare interface PlayerInfo {
    isLoaded(): boolean;
    getTrack(): string;
    getTitle(): string;
```

Reply:
No!
afaict this would remove the `.connect(handler)`! That is a deal breaker. Moreover, we're now unnecessarily duplicating APIs again!

Reply:
We need proper abstraction of the stable API and the experimental QML code. How to ensure this without a proper wrapper?
The other issue is that we need to prevent writing to internal strings in Mixxx.

Reply:
> How to ensure this without a proper wrapper?

`QmlPlayerProxy` is the wrapper.

>  The other issue is that we need to prevent writing to internal strings in Mixxx.

Again, what is "internal"? Everything exposed via `QmlPlayerProxy` should be freely readable and writable. There is no need for a second `QmlPlayerProxy` copy IMO.

Reply:
> We need proper abstraction of the stable API from the experimental QML code.

Well, the code has been around long enough. Lets make it stable then... The QML controller screen feature already assumes its stable.